README.md

```diff
@@ -0,0 +1,98 @@
+# [Latent State Models of Training Dynamics](https://arxiv.org/abs/2308.09543)
+
+Directly model training dynamics, then interpret the dynamics model.
+
+# Usage
+
+All these commands run from the main directory. If you want to run within a directory, you may need to change some imports.
+
+## Step 0: Collect training runs.
+You'll need to run these commands several times with different random seeds.
+
+Modular addition, one layer transformer:
+```bash
+python -m scripts.00_training.modular_addition --output_dir $output_dir --seed $seed 
+```
+
+Modular addition, one layer transformer with layer norm:
+```bash
+python -m scripts.00_training.modular_addition --output_dir $output_dir --seed $seed --use_ln
+```
+
+Sparse parities, MLP:
+```bash
+python -m scripts.00_training.sparse_parities --output_dir $output_dir --seed $seed 
+```
+
+MNIST, MLP:
+```bash
+python -m scripts.00_training.computer_vision --output_dir $output_dir --seed $seed --model_name  mlp --dataset_name mnist
+```
+
+CIFAR-100, ResNet18:
+```bash
+python -m scripts.00_training.computer_vision --output_dir $output_dir --seed $seed --model_name resnet18 --dataset_name cifar100 --use_batch_norm --use_residual
+```
+
+CIFAR-100, ResNet18 without batch norm or residual connections:
+```bash
+python -m scripts.00_training.computer_vision --output_dir $output_dir --seed $seed --model_name resnet18 --dataset_name cifar100
+```
+
+## Step 1 (optional): Calculate additional stats.
+
+The training scripts in this repo collect stats automatically, but you may need to run this when using downloaded checkpoints, such as those from [Pythia](https://github.com/EleutherAI/pythia) or [MultiBERTs](https://arxiv.org/abs/2106.16163).
+```bash
+pythom -m scripts.01_checkpoints_to_stats.compute_stats --model_dir <path/to/downloaded/models> --out_dir $out_dir
+```
+
+## Step 2: Collate statistics into 1 file.
+
+Take the stats computed in step 0 or step 1 and organize them into CSVs suitable for training the HMM. `--has_loss` is for cases. `--exp_type` exists to handle the logging of hyperparameters; see code.
+```bash
```

Comment:
You should specify what the valid strings are for `exp_type`. It's generally best practice for all the experiments to be reproducible from looking only at the `README`. The users should only have to look further into the code if they're planning on making changes